microsoft/PowerToys
# [Shortcut Guide] Activate with Windows key press

## Summary of the Pull Request

**What is this about:**
By community request, this re-enables the use of pressing the Windows key for showing the Shortcut Guide.
This is achievable by activating a setting to use the Windows key instead of the activation shortcut.
Changes were made to the centralized keyboard hook so we could support this behavior while maintaining Shortcut Guide in a different process that is only started when Shortcut Guide is activated.

![image](https://user-images.githubusercontent.com/26118718/134254130-b2c31a82-ab09-4e62-83be-80e3ea45e134.png)

**What is included in the PR:** 
- New settings in the Shortcut Guide to activate the "press the Windows key" behavior.
- Add logic to detect pressed keys to the centralized keyboard hook.
- Add functions specific to the PowerToys module interface just for the Shortcut Guide module. Needs to be generalized if we want more modules using this behavior in the future, which is not a goal for now.

**How does someone test / validate:** 
Play with the settings and verify the Shortcut Guide appears when you press the Windows key.

## Quality Checklist

- [x] **Linked issue:** #11963
